pkg/common-controller/snapshot_update_test.go

```diff
@@ -318,36 +318,37 @@ func TestSync(t *testing.T) {
 			initialClaims:     newClaimArray("claim5-3", "pvc-uid5-3", "1Gi", "volume5-3", v1.ClaimBound, &classEmpty),
 			initialVolumes:    newVolumeArray("volume5-3", "pv-uid5-3", "pv-handle5-3", "1Gi", "pvc-uid5-3", "claim5-3", v1.VolumeBound, v1.PersistentVolumeReclaimDelete, classEmpty),
 			initialSecrets:    []*v1.Secret{secret()},
 			expectSuccess:     true,
-			test:              testSyncContent,
+			test:              testSyncSnapshot,
 		},
 		{
 			name:              "5-4 - (dynamic) snapshot deletion candidate fail to mark for deletion due to failed API call",
-			initialSnapshots:  newSnapshotArray("snap5-4", "snapuid5-4", "claim5-4", "", validSecretClass, "snapcontent-snapuid5-4", &False, nil, nil, nil, false, true, &timeNowMetav1),
-			expectedSnapshots: newSnapshotArray("snap5-4", "snapuid5-4", "claim5-4", "", validSecretClass, "snapcontent-snapuid5-4", &False, nil, nil, nil, false, true, &timeNowMetav1),
+			initialSnapshots:  withSnapshotFinalizers(newSnapshotArray("snap5-4", "snapuid5-4", "claim5-4", "", validSecretClass, "snapcontent-snapuid5-4", &False, nil, nil, nil, false, false, &timeNowMetav1), utils.VolumeSnapshotBoundFinalizer),
+			expectedSnapshots: withSnapshotFinalizers(newSnapshotArray("snap5-4", "snapuid5-4", "claim5-4", "", validSecretClass, "snapcontent-snapuid5-4", &False, nil, nil, nil, false, false, &timeNowMetav1), utils.VolumeSnapshotBoundFinalizer),
 			initialContents:   newContentArray("snapcontent-snapuid5-4", "snapuid5-4", "snap5-4", "sid5-4", validSecretClass, "", "pv-handle5-4", deletionPolicy, nil, nil, true),
-			// result of the test framework - annotation is still set in memory, but update call fails.
+			// result of the test framework - annotation is still set in memory, but patch call fails.
 			expectedContents: withContentAnnotations(newContentArray("snapcontent-snapuid5-4", "snapuid5-4", "snap5-4", "sid5-4", validSecretClass, "", "pv-handle5-4", deletionPolicy, nil, nil, true), map[string]string{utils.AnnVolumeSnapshotBeingDeleted: "yes"}),
 			initialClaims:    newClaimArray("claim5-4", "pvc-uid5-4", "1Gi", "volume5-4", v1.ClaimBound, &classEmpty),
 			initialVolumes:   newVolumeArray("volume5-4", "pv-uid5-4", "pv-handle5-4", "1Gi", "pvc-uid5-4", "claim5-4", v1.VolumeBound, v1.PersistentVolumeReclaimDelete, classEmpty),
 			initialSecrets:   []*v1.Secret{secret()},
 			errors: []reactorError{
-				// Inject error to the forth client.VolumesnapshotV1().VolumeSnapshots().Update call.
-				{"update", "volumesnapshotcontents", errors.New("mock update error")},
+				// Inject error to the forth client.VolumesnapshotV1().VolumeSnapshotContents().Patch call.
+				{"patch", "volumesnapshotcontents", errors.New("mock update error")},
 			},
-			expectSuccess: false,
-			test:          testSyncContentError,
+			expectSuccess: true,
```

Comment:
I think we should always do `expectSuccess: true`? Specify false means don't check for failure, not asserting error is returned. But that is a big project. Let's leave it to another PR.